### PR TITLE
fix: #10628 expanduser path in TrainingArguments

### PR DESCRIPTION
**Pull request by PaulLerner**
## Who can review?

- trainer: @sgugger

### CODE REVIEW

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -507,6 +507,12 @@ class TrainingArguments:
     _n_gpu: int = field(init=False, repr=False, default=-1)
 
     def __post_init__(self):
+        # expand paths, see #10628
```
Comment (sgugger) [inline src/transformers/training_args.py:510]: A link on the full issue would be better, or a short description even better.

**Review by sgugger — APPROVED** [review]
Thanks a lot for fixing this issue!

File: src/transformers/training_args.py (hunk 1 above)
Comment (PaulLerner) [reply src/transformers/training_args.py]: Better now ?

Comment (sgugger) [reply src/transformers/training_args.py]: Yes, make sure to run `make style` on your branch to fix the styling issue and we should be good to merge!

Comment (PaulLerner) [reply src/transformers/training_args.py]: this gives a lot more diff than expected !
see https://github.com/PaulLerner/transformers/commit/e7fc1eb55c4a3ee5189b688a7011d9efaf099336

Comment (sgugger) [reply src/transformers/training_args.py]: Are you sure you have the right versions pinned (normally `pip install -e .[dev]` will install those for you). Anyhow can push the styling on your branch if it doesn't work.

**Comment by sgugger** [comment]
Thanks a lot for fixing that issue!

File: src/transformers/training_args.py (hunk 1 above)
Comment (PaulLerner) [reply src/transformers/training_args.py]: right I've only installed the pip release, duly noted for a future PR :)

**Pull request merged**